test/EndToEndTests/Tests/Client/Microsoft.OData.Client.E2E.Tests/ActionOverloadingTests/Server/PeopleController.cs

```diff
@@ -0,0 +1,124 @@
+﻿//-----------------------------------------------------------------------------
+// <copyright file="PeopleController.cs" company=".NET Foundation">
+//      Copyright (c) .NET Foundation and Contributors. All rights reserved.
+//      See License.txt in the project root for license information.
+// </copyright>
+//------------------------------------------------------------------------------
+
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.OData.Formatter;
+using Microsoft.AspNetCore.OData.Query;
+using Microsoft.AspNetCore.OData.Routing.Controllers;
+using Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd;
+
+namespace Microsoft.OData.Client.E2E.Tests.ActionOverloadingTests.Server
+{
+    public class PeopleController : ODataController
+    {
+        public PeopleController() { }
+
+        [EnableQuery]
+        [HttpGet("odata/People")]
+        public IActionResult Get()
+        {
+            var people = CommonEndToEndDataSource.People;
+            return Ok(people);
+        }
+
+        [EnableQuery]
+        [HttpGet("odata/People({key})")]
+        public IActionResult Get(int key)
+        {
+            var person = CommonEndToEndDataSource.People?.FirstOrDefault(a => a.PersonId == key);
+            return Ok(person);
+        }
+
+        [HttpPost("odata/Default.UpdatePersonInfo")]
+        public IActionResult UpdatePersonInfo()
+        {
+            CommonEndToEndDataSource.People.First().Name += "[UpdataPersonName]";
+            return Ok();
+        }
+
+        [HttpPost("odata/People({key})/Default.UpdatePersonInfo")]
+        public IActionResult UpdatePersonInfo([FromODataUri] int key)
+        {
+            CommonEndToEndDataSource.People.First(a => a.PersonId == key).Name += "[UpdataPersonName]";
+            return Ok();
+        }
+
+        [HttpPost("odata/People({key})/Microsoft.OData.Client.E2E.Tests.Common.Server.EndToEnd.Person/Default.UpdatePersonInfo")]
+        public IActionResult UpdateEmployeeInfo([FromODataUri] int key)
+        {
+            if (!ModelState.IsValid)
+            {
+                return BadRequest();
+            }
+
+            CommonEndToEndDataSource.People.First(a => a.PersonId == key).Name += "[UpdataPersonName]";

```

Comment:
This will throw an unhandled exception if the entity by the key doesn't exist. Why not throw an exception instead?

Reply:
Doesn't seem like the unhandled exception when the entity has not been found has been addressed. Should return a 404 if the person is not found?